shell/platform/android/test/io/flutter/embedding/engine/FlutterEnginePluginRegistryTest.java

```diff
@@ -58,6 +65,43 @@ public void itDoesNotRegisterTheSamePluginTwice() {
     assertEquals(0, fakePlugin2.detachmentCallCount);
   }
 
+  @Test
+  public void activityResultListenerCanBeRemovedFromListener() {
```

Comment:
I don't understand the name of this test.  I don't see any removal happening.

edit:  I read through a bit more and I saw that you are removing the listener in `FakeActivityResultListener`.  This test would be easier to follow if you used an anonymous class.  Shouldn't this test try to trigger the listener again to show that it was successfully removed?

Reply:
The purpose of the test is to make sure that removing listener when it is invoked does not break the iteration over listeners to invoke.
Before the change, removing a listener (which is not the last one to invoke) would trigger a ConcurrentModificationException because a listener was removed during an iteration of the listeners collection.
The problem is not the test's implemention but its name.
I'm not sure I have a proper name to suggest, something like: "activity result listeners continue to be triggered after a listener removal"?

Reply:
Or maybe something like "result listener self removal does not break listeners invocation"

Reply:
In attention to NoamDev, I could use an anonymus implementation, but I would need to implement the code twice, so I decided to create a class and I use two instances.

The interesting part that the first called listener must remove itself to break the iterator.

Regarding the name I can change it to anything you'd like, but I had no better idea.

Reply:
After reading your message again @gaaclarke yeah makes sense to call it a second time to verify that the listener was removed. My test was more about that you can remove the listener without a crash. I'll extend the test (and fix a typo I found)